defaults/json.go

```diff
@@ -10,19 +10,24 @@ import (
 // This function will try to read from given path, if it is empty will try
 // fallback path. If it fails on fallback, it will set value to it's defaults
 func ReadFromJSON(path, fallbackPath string, value interface{}) error {
+	if err := Set(value); err != nil {
+		return err
+	}
+
 	if path != "" {
 		if err := readJSON(path, &value); err != nil {
-			return err
+			return fmt.Errorf("failed to read from path %s: %w", path, err)
 		}
 		return nil
 	}
 
-	if err := readJSON(fallbackPath, value); err == nil {
+	// If the fallback path doesn't exist, return.
+	if _, err := os.Stat(fallbackPath); err != nil && os.IsNotExist(err) {
 		return nil
 	}
 
-	if err := Set(value); err != nil {
-		return err
+	if err := readJSON(fallbackPath, value); err != nil {
+		return fmt.Errorf("failed to read from path %s: %w", fallbackPath, err)
```

Comment:
Nit to make the error slightly more useful
```suggestion
		return fmt.Errorf("failed to read from fallback path %s: %w", fallbackPath, err)
```